### ISSUE TITLE
Query makedepends option.

### ISSUE THREAD

**Issue by bartoszek** [archlinuxfr/package-query#131]
Would be useful to have --qmakedepends option to filter out packages installed when building some AUR packages (stuff like: asciidoc, automoc, apache-ant etc..). That how you could filter those out when removing 'truly'  unneeded packages, preserving makedeps for later AUR upgrades.

**Comment by f2404** [comment]
A good idea! Could be done relatively easy by parsing PKGBUILD file.

**Comment by f2404** [comment]
Well, I think I need to understand your request better. What would you like to see in the output?
There's already an option to query makedepends for a given AUR package: %M format flag. This was my first thought.
However, you could be looking for a list of all packages installed as makedepends. Is this your goal?

**Comment by f2404** [comment]
@bartoszek ping

**Comment by bartoszek** [comment]
> looking for a list of all packages installed as makedepends
Yes that was my point. While clearing my system form unneeded packages (pacman -Qqdt) I find that most of those packages are mkdeps of some AUR packages which would need those when upgrading. My solution was to make the list of all mkdeps of all my alien packages:
`for pkg in $(pacman -Qqm) ; do package-query -A -f '%M' $pkg|sed -e 's:^-$::' -e 's: :\n:g' ; done|sort -u`
and filter those out.
Than I thought that 'package-query' could have additional switch like '--qmakedepends' which would query AUR for packages that list target as makedependency.

**Comment by f2404** [comment]
Thanks, got you.
On the other hand: is it worth adding another option into a program if one can achieve the same result with a rather clean combination of existing calls?

**Comment by InfernoZeus** [comment]
@bartoszek thanks for that command. I took it and added some functions around it to provide a single function to delete all non-makedepends non-optdepends orphans:
`
listmakedepends() {
        for pkg in $(pacman -Qqm); do
                package-query -A -f '%M' $pkg|sed -e 's:^-$::' -e 's: :\n:g'
        done | sort -u
}
listallorphans() {
        pacman -Qdtq
}
listorphans() {
        comm -23 <(listallorphans) <(listmakedepends)
}
removeorphans() {
        pacman -Rs $(listorphans)
}
`
